### PR TITLE
Feature/chat

### PR DESCRIPTION
**Pull request by alonzopt**
Chat resources and collections, plus tests.
This is a branch off of feature/paging and so can be merged instead or will be updated after feature/paging is merged.

### CODE REVIEW

**Comment by samamorgan** [comment]
@alonzopt I'll wait to review this one until feature/paging is merged.

File: welkin/models/chat.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        return f"{self.sender['clientType']} {self.message}"
+
+
+class Chats(Collection):
+    resource = Chat
+    iterator = MetaIterator
+
+    def get(
+        self,
+        pageSize: int = 20,
```
Comment (samamorgan) [inline welkin/models/chat.py:21]: These arguments don't follow [PEP8 conventions](https://peps.python.org/pep-0008/#function-and-method-arguments) (should be snake_case).

The only time we should be using lowerCamelCase is when we're interacting with data that was parsed from a request (like `Encounter().patientId`). If it's purely python functionality, snake_case is correct (like `Encounter().get(patient_id="")`.

File: welkin/models/chat.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        includeArchived: bool = False,
+        pageToken: str = None,
+    ):
+        params = {
+            "includeArchived": includeArchived,
+        }
+        # pageToken and pageSize are mutually exclusive
+        if pageToken:
+            params["pageToken"] = pageToken
```
Comment (samamorgan) [inline welkin/models/chat.py:30]: Pagination parameters should be handled by the underlying paginator class. A user should be able to pass `paginate=True` and `size=int` to any paginated collection get method, those would pass to the request method through args/kwargs, and that in turn passes to the paginator class.

File: welkin/models/chat.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        query: str = None,
+        pageToken: str = None,
+        pageSize: int = 20,
+        contentPageSize: int = 20,
+        includeArchived: bool = False,
+    ):
+        if not query and not pageToken:
+            raise Exception('"query" or "pageToken" required.')
+        params = {
+            "pageSize": pageSize,
```
Comment (samamorgan) [inline welkin/models/chat.py:58]: Same comment here as above on pagination handling

**Review by samamorgan — CHANGES_REQUESTED** [review]
Just a few changes needed. I think some restructuring of `MetaIterator` might be required. I'm hoping we didn't lose intended functionality in #23

**Comment by alonzopt** [comment]
Requested changes complete. Got the variables renamed and moved code out of Chat Collections and into the MetaIterator

File: welkin/models/chat.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class SearchChats(Collection):
+    resource = ChatSearchResult
+    iterator = MetaIterator
+
+    def get(
+        self,
+        query: str,
+        content_page_size: int = 20,
```
Comment (samamorgan) [inline welkin/models/chat.py:42]: So chat search has an additional parameter for controlling pagination? Very similar to the token-based pagination, but one additional parameter? That's the 5th scheme. Silly.

**Review by samamorgan — APPROVED** [review]
LGTM! I'm going to add a quick line to one of the tests to check the string representation of a chat so we cover that line.

**Pull request merged**